Tests failed: Wasm.Build.Tests.BlazorWasmBuildPublishTests due to SIGKILL

Failing tests: Wasm.Build.Tests.BlazorWasmBuildPublishTests due to SIGKILL

Affected tests:
- WithNativeReference_AOTInProjectFile -- `Wasm.Build.Tests.BlazorWasmBuildPublishTests.WithNativeReference_AOTInProjectFile`
- WithNativeReference_AOTOnCommandLine -- `Wasm.Build.Tests.BlazorWasmBuildPublishTests.WithNativeReference_AOTOnCommandLine`
- NativeBuild_WithDeployOnBuild_UsedByVS -- `Wasm.Build.Tests.BlazorWasmBuildPublishTests.NativeBuild_WithDeployOnBuild_UsedByVS`

Failures as of 6/20:
- WithNativeReference_AOTInProjectFile
    - 16 failures (3 in Rolling) in [last 7 days](https://runfo.azurewebsites.net/search/tests/?q=started%3A~7%20definition%3Aruntime%20name%3A%22wasm.build.tests.blazorwasmbuildpublishtests.withnativereference_aotinprojectfile%22) / 42 failures (4 in Rolling) in [last 30 days](https://runfo.azurewebsites.net/search/tests/?q=started%3A%7E30+definition%3Aruntime+name%3A%22wasm.build.tests.blazorwasmbuildpublishtests.withnativereference_aotinprojectfile%22)
- WithNativeReference_AOTOnCommandLine
    - 16 failures (3 in Rolling) in [last 7 days](https://runfo.azurewebsites.net/search/tests/?q=started%3A~7%20definition%3Aruntime%20name%3A%22wasm.build.tests.blazorwasmbuildpublishtests.withnativereference_aotoncommandline%22) / 41 failures (4 in Rolling) in [last 30 days](https://runfo.azurewebsites.net/search/tests/?q=started%3A%7E30+definition%3Aruntime+name%3A%22wasm.build.tests.blazorwasmbuildpublishtests.withnativereference_aotoncommandline%22)
- NativeBuild_WithDeployOnBuild_UsedByVS
    - 25 failures (2 in Rolling) in [last 7 days](https://runfo.azurewebsites.net/search/tests/?q=started%3A~7%20definition%3Aruntime%20name%3A%22wasm.build.tests.blazorwasmbuildpublishtests.nativebuild_withdeployonbuild_usedbyvs%22) / 50 failures (4 in Rolling) in [last 30 days](https://runfo.azurewebsites.net/search/tests/?q=started%3A%7E30+definition%3Aruntime+name%3A%22wasm.build.tests.blazorwasmbuildpublishtests.nativebuild_withdeployonbuild_usedbyvs%22)

Platform: net7.0-Browser-Release-wasm-Mono_Release_Linux-BuildWasmApps-Ubuntu.1804.Amd64.Open

Output example:
```
Wasm.Build.Tests.BlazorWasmBuildPublishTests.WithNativeReference_AOTOnCommandLine:

[] emcc : error : '/datadisks/disk1/work/B58909DD/w/A8850950/e/dotnet-workload/packs/Microsoft.NET.Runtime.Emscripten.3.1.7.Sdk.linux-x64/7.0.0-preview.6.22281.1/tools/bin/wasm-emscripten-finalize -g --dyncalls-i64 --dwarf /datadisks/disk1/work/B58909DD/w/A8850950/e/blz_nativeref_aot_Debug/obj/Debug/net7.0/wasm/for-publish/dotnet.wasm -o /datadisks/disk1/work/B58909DD/w/A8850950/e/blz_nativeref_aot_Debug/obj/Debug/net7.0/wasm/for-publish/dotnet.wasm --detect-features' failed (received SIGKILL (-9)) [/datadisks/disk1/work/B58909DD/w/A8850950/e/blz_nativeref_aot_Debug/blz_nativeref_aot_Debug.csproj]
[] /datadisks/disk1/work/B58909DD/w/A8850950/e/dotnet-workload/packs/Microsoft.NET.Runtime.WebAssembly.Sdk/7.0.0-ci/Sdk/WasmApp.Native.targets(424,5): error MSB3073: The command "emcc "@/datadisks/disk1/work/B58909DD/w/A8850950/e/dotnet-workload/packs/Microsoft.NETCore.App.Runtime.Mono.browser-wasm/7.0.0-ci/runtimes/browser-wasm/native/src/emcc-default.rsp" "@/datadisks/disk1/work/B58909DD/w/A8850950/e/dotnet-workload/packs/Microsoft.NETCore.App.Runtime.Mono.browser-wasm/7.0.0-ci/runtimes/browser-wasm/native/src/emcc-link.rsp" "@/datadisks/disk1/work/B58909DD/w/A8850950/e/blz_nativeref_aot_Debug/obj/Debug/net7.0/wasm/for-publish/emcc-link.rsp"" exited with code 1. [/datadisks/disk1/work/B58909DD/w/A8850950/e/blz_nativeref_aot_Debug/blz_nativeref_aot_Debug.csproj]
```
<!--Known issue error report start -->
### Report
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|0|
<!--Known issue error report end -->

--- a/src/mono/wasm/Wasm.Build.Tests/Blazor/BuildPublishTests.cs
+++ b/src/mono/wasm/Wasm.Build.Tests/Blazor/BuildPublishTests.cs
@@ -84,7 +84,7 @@ public class BuildPublishTests : BuildTestBase
     [Theory]
     [InlineData("Debug")]
     [InlineData("Release")]
-    public void WithDllImportInMainAssembly(string config)
+    public async Task WithDllImportInMainAssembly(string config)
     {
         // Based on https://github.com/dotnet/runtime/issues/59255
         string id = $"blz_dllimp_{config}";
@@ -124,12 +124,12 @@ public class BuildPublishTests : BuildTestBase
         BlazorPublish(new BlazorBuildOptions(id, config, NativeFilesType.Relinked));
         CheckNativeFileLinked(forPublish: true);
 
-        // [ActiveIssue("https://github.com/dotnet/runtime/issues/79514")]
-        //await BlazorRun(config, async (page) => {
-            //await page.Locator("text=\"cpp_add\"").ClickAsync();
-            //var txt = await page.Locator("p[role='test']").InnerHTMLAsync();
-            //Assert.Equal("Output: 22", txt);
-        //});
+        await BlazorRun(config, async (page) =>
+        {
+            await page.Locator("text=\"cpp_add\"").ClickAsync();
+            var txt = await page.Locator("p[role='test']").InnerHTMLAsync();
+            Assert.Equal("Output: 22", txt);
+        });
 
         void CheckNativeFileLinked(bool forPublish)
         {

--- a/src/mono/wasm/Wasm.Build.Tests/Blazor/MiscTests.cs
+++ b/src/mono/wasm/Wasm.Build.Tests/Blazor/MiscTests.cs
@@ -30,7 +30,9 @@ public class MiscTests : BuildTestBase
                                     ? ("<EmccLinkOptimizationFlag>-O1</EmccLinkOptimizationFlag>" +
                                         "<EmccCompileOptimizationFlag>-O1</EmccCompileOptimizationFlag>")
                                     : string.Empty;
-        AddItemsPropertiesToProject(projectFile, extraProperties: nativeRelink ? string.Empty : "<RunAOTCompilation>true</RunAOTCompilation>");
+        if (!nativeRelink)
+            extraProperties += "<RunAOTCompilation>true</RunAOTCompilation>";
+        AddItemsPropertiesToProject(projectFile, extraProperties: extraProperties);
 
         // build with -p:DeployOnBuild=true, and that will trigger a publish
         (CommandResult res, _) = BuildInternal(id, config, publish: false, setWasmDevel: false, "-p:DeployOnBuild=true");

--- a/src/mono/wasm/Wasm.Build.Tests/Blazor/MiscTests.cs
+++ b/src/mono/wasm/Wasm.Build.Tests/Blazor/MiscTests.cs
@@ -26,6 +26,10 @@ public class MiscTests : BuildTestBase
     {
         string id = $"blz_deploy_on_build_{config}_{nativeRelink}_{Path.GetRandomFileName()}";
         string projectFile = CreateProjectWithNativeReference(id);
+        string extraProperties = config == "Debug"
+                                    ? ("<EmccLinkOptimizationFlag>-O1</EmccLinkOptimizationFlag>" +
+                                        "<EmccCompileOptimizationFlag>-O1</EmccCompileOptimizationFlag>")
+                                    : string.Empty;
         AddItemsPropertiesToProject(projectFile, extraProperties: nativeRelink ? string.Empty : "<RunAOTCompilation>true</RunAOTCompilation>");
 
         // build with -p:DeployOnBuild=true, and that will trigger a publish

--- a/src/mono/wasm/Wasm.Build.Tests/Blazor/MiscTests.cs
+++ b/src/mono/wasm/Wasm.Build.Tests/Blazor/MiscTests.cs
@@ -22,7 +22,6 @@ public class MiscTests : BuildTestBase
     [InlineData("Debug", false)]
     [InlineData("Release", true)]
     [InlineData("Release", false)]
-    //[ActiveIssue("https://github.com/dotnet/runtime/issues/70985", TestPlatforms.Linux)]
     public void NativeBuild_WithDeployOnBuild_UsedByVS(string config, bool nativeRelink)
     {
         string id = $"blz_deploy_on_build_{config}_{nativeRelink}_{Path.GetRandomFileName()}";
@@ -67,7 +66,12 @@ public class MiscTests : BuildTestBase
     {
         string id = $"blz_aot_prj_file_{config}_{Path.GetRandomFileName()}";
         string projectFile = CreateBlazorWasmTemplateProject(id);
-        AddItemsPropertiesToProject(projectFile, extraProperties: "<RunAOTCompilation>true</RunAOTCompilation>");
+
+        string extraProperties = config == "Debug"
+                                    ? ("<EmccLinkOptimizationFlag>-O1</EmccLinkOptimizationFlag>" +
+                                        "<EmccCompileOptimizationFlag>-O1</EmccCompileOptimizationFlag>")
+                                    : string.Empty;
+        AddItemsPropertiesToProject(projectFile, extraProperties: "<RunAOTCompilation>true</RunAOTCompilation>" + extraProperties);
 
         // No relinking, no AOT
         BlazorBuild(new BlazorBuildOptions(id, config, NativeFilesType.FromRuntimePack));

--- a/src/mono/wasm/Wasm.Build.Tests/Blazor/NativeRefTests.cs
+++ b/src/mono/wasm/Wasm.Build.Tests/Blazor/NativeRefTests.cs
@@ -20,12 +20,15 @@ public class NativeRefTests : BuildTestBase
     [Theory]
     [InlineData("Debug")]
     [InlineData("Release")]
-    [ActiveIssue("https://github.com/dotnet/runtime/issues/70985", TestPlatforms.Linux)]
     public void WithNativeReference_AOTInProjectFile(string config)
     {
         string id = $"blz_nativeref_aot_{config}_{Path.GetRandomFileName()}";
         string projectFile = CreateProjectWithNativeReference(id);
-        AddItemsPropertiesToProject(projectFile, extraProperties: "<RunAOTCompilation>true</RunAOTCompilation>");
+        string extraProperties = config == "Debug"
+                                    ? ("<EmccLinkOptimizationFlag>-O1</EmccLinkOptimizationFlag>" +
+                                        "<EmccCompileOptimizationFlag>-O1</EmccCompileOptimizationFlag>")
+                                    : string.Empty;
+        AddItemsPropertiesToProject(projectFile, extraProperties: "<RunAOTCompilation>true</RunAOTCompilation>" + extraProperties);
 
         BlazorBuild(new BlazorBuildOptions(id, config, NativeFilesType.Relinked));
 
@@ -38,11 +41,15 @@ public class NativeRefTests : BuildTestBase
     [Theory]
     [InlineData("Debug")]
     [InlineData("Release")]
-    [ActiveIssue("https://github.com/dotnet/runtime/issues/70985", TestPlatforms.Linux)]
     public void WithNativeReference_AOTOnCommandLine(string config)
     {
         string id = $"blz_nativeref_aot_{config}_{Path.GetRandomFileName()}";
-        CreateProjectWithNativeReference(id);
+        string projectFile = CreateProjectWithNativeReference(id);
+        string extraProperties = config == "Debug"
+                                    ? ("<EmccLinkOptimizationFlag>-O1</EmccLinkOptimizationFlag>" +
+                                        "<EmccCompileOptimizationFlag>-O1</EmccCompileOptimizationFlag>")
+                                    : string.Empty;
+        AddItemsPropertiesToProject(projectFile, extraProperties: extraProperties);
 
         BlazorBuild(new BlazorBuildOptions(id, config, NativeFilesType.Relinked));
 

--- a/src/mono/wasm/Wasm.Build.Tests/BuildTestBase.cs
+++ b/src/mono/wasm/Wasm.Build.Tests/BuildTestBase.cs
@@ -29,7 +29,7 @@ namespace Wasm.Build.Tests
     public abstract class BuildTestBase : IClassFixture<SharedBuildPerTestClassFixture>, IDisposable
     {
         public const string DefaultTargetFramework = "net8.0";
-        public const string DefaultTargetFrameworkForBlazor = "net7.0";
+        public const string DefaultTargetFrameworkForBlazor = "net8.0";
         protected static readonly bool s_skipProjectCleanup;
         protected static readonly string s_xharnessRunnerCommand;
         protected string? _projectDir;

--- a/src/mono/wasm/Wasm.Build.Tests/WasmTemplateTests.cs
+++ b/src/mono/wasm/Wasm.Build.Tests/WasmTemplateTests.cs
@@ -416,7 +416,6 @@ namespace Wasm.Build.Tests
         }
 
         [ConditionalFact(typeof(BuildTestBase), nameof(IsUsingWorkloads))]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/79514")]
         public async Task BlazorRunTest()
         {
             string config = "Debug";

